docs/root/configuration/listeners/network_filters/postgres_proxy_filter.rst

```diff
@@ -0,0 +1,92 @@
+.. _config_network_filters_postgres_proxy:
+
+Postgres proxy
+================
+
+The Postgres proxy filter decodes the wire protocol between a Postgres client (downstream) and a Postgres server
+(upstream). The decoded information is currently used only to produce Postgres level statistics like sesions,
+statements or transactions executed, among others. This current version does not decode SQL queries. Future versions may
+add more statistics and more advanced capabilities. When the Postgres filter detects that a session is encrypted, the messages are ignored and no decoding takes
+place. More information:
+
+* Postgres :ref:`architecture overview <arch_overview_postgres>`
+
+.. attention::
+
+   The `postgres_proxy` filter is experimental and is currently under active development.
+   Capabilities will be expanded over time and the configuration structures are likely to change.
+
+
+.. warning::
+
+   The `postgreql_proxy` filter was tested only with
+   `Postgres frontend/backend protocol version 3.0`_, which was introduced in
+   Postgres 7.4. Earlier versions are thus not supported. Testing is limited
+   anyway to not EOL-ed versions.
+
+   .. _Postgres frontend/backend protocol version 3.0: https://www.postgresql.org/docs/current/protocol.html
+
+
+
+Configuration
+-------------
+
+The Postgres proxy filter should be chained with the TCP proxy as shown in the configuration
+example below:
+
+.. code-block:: yaml
+
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.postgres_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.config.filter.network.postgres_proxy.v3alpha.PostgresProxy
```

Comment:
You have `envoy.config` here and below. Can you update to the v3 layout `envoy.extensions`?

Reply:
Corrected to match v3 layout.